### PR TITLE
Missing references test

### PR DESCRIPTION
**Pull request by mildas**
#### Description:
Add test for checking missing references in profiles.

Usage:
`$ tests/missing_refs.sh python3 build/ssg-rhel8-xccdf.xml ospp stig hipaa`

$1 - python executable (for cmake, `${PYTHON_EXECUTABLE}`)
$2 - product xccdf
$3+ - list of profiles

#### Rationale:
Test that will be added to gating.

### CODE REVIEW

**Comment by mildas** [comment]
List of rules missing references, output from the test:
```55/73 Test #56: missing-references-ssg-rhel7-xccdf.xml ..............................***Failed    1.64 sec
*** rules of 'stig' profile missing STIG IDs: 241 of 252 have them [4% missing]
   audit_rules_login_events_tallylog            
   audit_rules_privileged_commands_sudoedit     
   configure_firewalld_rate_limiting               dconf_db_up_to_date                          
   dconf_gnome_disable_automount                
   dconf_gnome_disable_automount_open           
   dconf_gnome_disable_autorun                  
   mount_option_dev_shm_nodev                   
   mount_option_dev_shm_nosuid                     package_aide_installed                       
   xwindows_runlevel_target                     
*** rules of 'hipaa' profile missing HIPAA Refs: 142 of 143 have them [0% missing]
   dconf_db_up_to_date            

68/73 Test #69: missing-references-ssg-rhel8-xccdf.xml ..............................***Failed    1.37 sec
*** rules of 'hipaa' profile missing HIPAA Refs: 134 of 137 have them [2% missing]
   configure_crypto_policy                      
   configure_ssh_crypto_policy                     dconf_db_up_to_date          
```
@ggbecker Is the rhel7 stig doable? Or is it same as you mentioned in https://github.com/ComplianceAsCode/content/issues/6844#issuecomment-820429384 and I should exclude it for now and create issue?

**Comment by mildas** [comment]
I will split this PR into two PRs - this one will add only the test and the new PR will add it to `SSGCommon.cmake`

**Comment by jan-cerny** [comment]
Have you considered making this an CMake macro instead of a bash script to avoid having multiple abstraction levels?

File: tests/missing_refs.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+PYTHON_EXECUTABLE="$1"
+BENCHMARK="$2"
+PROFILES="$3"
```
Comment (jan-cerny) [inline tests/missing_refs.sh:4]: this is a risk of quoting problems

what about making every argument starting from 3 a profile? eg. use shift

Comment (mildas) [reply tests/missing_refs.sh:4]: Aha, that's much better approach. Thanks, I will update it

**Comment by mildas** [comment]
@jan-cerny No. Actually, I didn't think about CMake macro approach at all because it didn't occur to me.
With the macro, we could avoid multiple abstraction levels, but do you think it will be readable? I mean, if implementing this in CMake macro won't be unnecessarily complicated when it can be easily done in Bash script + other tests are in a form of standalone script and this one will be as a macro.

**Comment by jan-cerny** [comment]
I think it would be readable

> other tests are in a form of standalone script and this one will be as a macro.

That's a good argument against it.

**Comment by mildas** [comment]
@jan-cerny What do you think? Should I implement it as CMake macro or are you fine with current approach?

**Comment by jan-cerny** [comment]
@mildas I'm fine with either way.

**Pull request merged**